src/main/groovy/org/ajoberstar/gradle/git/publish/GitPublishPlugin.groovy

```diff
@@ -71,6 +71,10 @@ class GitPublishPlugin implements Plugin<Project> {
     task.with {
       group = 'publishing'
       description = 'Prepares a git repo for new content to be generated.'
+
+      // creates/clones the repo
+      outputs.dir(extension.repoDir)
```

Comment:
I guess you also should do `upToDateWhen { false }` to make sure the task is always executed. Given that is has no inputs that is what would happen if the output directory contents would not change. Alternatively, you could also add the inputs to the task (git commit hash, branch, remote?) if that is possible. Same goes for the other tasks in here.
In the future, we may provide you with a way to declare that there should not be any up-to-date checks for a task, so you don't pay the performance penalty of checking all the files in the repository for changes.